frontend/app/(defaultLayout)/components/GoogleAnalytics.tsx

```diff
@@ -1,53 +1,64 @@
-import { GA_DEBUG, GA_KEY, GA_URL, GTAG_ID } from '@config/env';
-import { usePathname, useSearchParams } from 'next/navigation';
-import Script from 'next/script';
+import { GA_URL, GTAG_ID, GA_KEY, GA_DEBUG } from '@config/env';
 import * as React from 'react';
+import Script from 'next/script';
+import Head from 'next/head';
+import { useRouter } from 'next/router';
 
 declare const ga: (command: string, hitType: string, url?: string) => void;
 
-const handleRouteChange = (url: string) => {
-   if (typeof ga !== 'undefined') {
-      ga('ifixit.set', 'page', url);
-      ga('ifixit.send', 'pageview');
-   }
-};
-
+// NOTE: GA4 is setup in _document.tsx
 export function GoogleAnalytics() {
-   const pathname = usePathname();
-   const searchParams = useSearchParams();
-
+   const router = useRouter();
    React.useEffect(() => {
-      const url = `${pathname}?${searchParams}`;
-      handleRouteChange(url);
-   }, [pathname, searchParams]);
+      const handleRouteChange = (url: string) => {
+         if (typeof ga !== 'undefined') {
+            ga('ifixit.set', 'page', url);
+            ga('ifixit.send', 'pageview');
+         }
+      };
+      router.events.on('routeChangeComplete', handleRouteChange);
+      router.events.on('hashChangeComplete', handleRouteChange);
+      return () => {
+         router.events.off('routeChangeComplete', handleRouteChange);
+         router.events.off('hashChangeComplete', handleRouteChange);
+      };
+   }, [router?.events]);
 
    const wantsUA = GA_URL && GA_KEY;
-   const wantsGA4 = Boolean(GTAG_ID);
-
+   const wantsGA4 = GTAG_ID;
    return (
       <>
-         {wantsUA && <UA />}
          {wantsGA4 && <GA4 />}
+         {wantsUA && <UA />}
       </>
    );
 }
 
 function GA4() {
+   const router = useRouter();
+   const { query } = router;
+
+   const debugMode = GA_DEBUG || query.ga4_debug === 'true';
    return (
       <>
+         <Head>
+            <script
+               id="gtag-ga4"
+               dangerouslySetInnerHTML={{
+                  __html: `
+               window.dataLayer = window.dataLayer || [];
+               function gtag(){dataLayer.push(arguments);}
+               gtag('js', new Date());
+               gtag('config', '${GTAG_ID}', ${debugMode} ? '{ debug_mode: true }' : '{}'
+                  );
+               `,
+               }}
+            />
+         </Head>
```

Comment:
There is no `<Head>` component in the app router. I think we need to use `<Script beforeInteractive>` here

Reply:
unfortunately, `beforeInteractive` isn't allowed in this file as it is not app/layout.tsx